### PR TITLE
Textual rewrite

### PR DESCRIPTION
**Pull request by dmoisset**
Follow up on #8

This reimplements the application with a Textual UI, adding line highlighting+automated scrolling

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,66 @@
-# codoscope
\ No newline at end of file
+# Codoscope
+
+Visualize the python compile pipeline
+
+Based in https://github.com/iritkatriel/codoscope/tree/main
```
Comment (iritkatriel) [inline README.md:5]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -1 +1,66 @@
...
+
+## Install
+
+This requires:
+
+* CPython 3.13 (tested with 3.13.0b1)
+* dependencies in requirements.txt
+  * You will have to manually install `tree-sitter-languages` manually to have syntax
```
Comment (iritkatriel) [inline README.md:13]:
```suggestion
  * You will have to manually install `tree-sitter-languages` to have syntax
```

File: src/bytecode_widget.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+if VERSION_3_13:
+    compiler_codegen: Any
+    optimize_cfg: Any
+    assemble_code_object: Any
+    from _testinternalcapi import compiler_codegen, optimize_cfg, assemble_code_object  # type: ignore
+else:
+
+    def _fail(*args: Any, **kwargs: Any) -> Any:
+        raise Exception("This function should never been have called")
```
Comment (iritkatriel) [inline src/bytecode_widget.py:21]:
```suggestion
        raise Exception("This function should never have been called")
```

File: src/source_widget.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+from events import HoverLine
+
+
+class SourceWidget(Container):
+
+    def compose(self) -> ComposeResult:
+        with VerticalScroll(classes="scroller"):
+            # To use an editor
+            # yield TextArea.code_editor("", language="python", classes="editor")
```
Comment (iritkatriel) [inline src/source_widget.py:18]: Did yo intend to delete this comment?

File: src/source_widget.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            yield Static(classes="editor", expand=True)
+
+    def set_code(self, code: str) -> None:
+        # To use an editor
+        # source = self.query_one(".editor", TextArea)
+        # source.text = code
```
Comment (iritkatriel) [inline src/source_widget.py:24]: And this one?

File: src/viewer.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+            ("7", "toggle_code_obj", "Final BC"),
+        ]
+
+    show_source = var(True)
+    show_tokens = var(True)
+    show_ast = var(False)
+    show_opt_ast = var(False)
+    show_pseudo_bc = var(False)
+    show_opt_pseudo_bc = var(False)
+    show_code_obj = var(False)
```
Comment (iritkatriel) [inline src/viewer.py:63]: Maybe start with show_source and show_code_obj True and the rest False?

**Comment by iritkatriel** [comment]
Let's remove my implementation in the same PR.

File: src/source_widget.py (hunk 4 above)
Comment (dmoisset) [reply src/source_widget.py]: Yes, thanks for the catch. I've removed both.

**Review by iritkatriel — APPROVED** [review]

**Comment by iritkatriel** [comment]
Thank you!

**Pull request merged**